tgui/packages/tgui/interfaces/CommunicationsConsole.js

```diff
@@ -0,0 +1,712 @@
+      // if one can be given
+      tooltip={canBuyShuttles !== 1 ? canBuyShuttles : undefined}
+      tooltipPosition="right"
+      onClick={() => act("setState", { state: STATE_BUYING_SHUTTLE })}
+    />);
+  }
+
+  if (canMessageAssociates) {
+    generalFunctions.push(<Button
+      icon="comment-o"
+      content={`Send message to ${emagged ? "[UNKNOWN]" : "CentCom"}`}
+      disabled={!importantActionReady}
+      onClick={() => setMessagingAssociates(true)}
+    />);
+
+    if (messagingAssociates) {
+      children.push(<MessageModal
+        label={`Message to transmit to ${emagged ? "[ABNORMAL ROUTING COORDINATES]" : "CentCom"} via quantum entanglement`}
+        notice="Please be aware that this process is very expensive, and abuse will lead to...termination. Transmission does not guarantee a response."
+        icon="bullhorn"
+        buttonText="Send"
+        onBack={() => setMessagingAssociates(false)}
+        onSubmit={message => {
+          setMessagingAssociates(false);
+          act("messageAssociates", {
+            message,
+          });
+        }}
+      />);
+    }
+  }
+
+  if (canRequestNuke) {
+    generalFunctions.push(<Button
+      icon="bomb"
+      content="Request Nuclear Authentication Codes"
+      disabled={!importantActionReady}
+      onClick={() => setRequestingNukeCodes(true)}
+    />);
+
+    if (requestingNukeCodes) {
+      children.push(<MessageModal
+        label="Reason for requesting nuclear self-destruct codes"
+        notice="Misuse of the nuclear request system will not be tolerated under any circumstances. Transmission does not guarantee a response."
+        icon="bomb"
+        buttonText="Request Codes"
+        onBack={() => setRequestingNukeCodes(false)}
+        onSubmit={reason => {
+          setRequestingNukeCodes(false);
+          act("requestNukeCodes", {
+            reason,
+          });
+        }}
+      />);
+    }
+  }
+
+  if (emagged) {
+    generalFunctions.push(<Button
+      icon="undo"
+      content="Restore Backup Routing Data"
+      onClick={() => act("restoreBackupRoutingData")}
+    />);
+  }
+
+  if (callingShuttle) {
+    children.push(<MessageModal
+      label="Nature of emergency"
+      icon="space-shuttle"
+      buttonText="Call Shuttle"
+      minLength={callShuttleReasonMinLength}
+      onBack={() => setCallingShuttle(false)}
+      onSubmit={reason => {
+        setCallingShuttle(false);
+        act("callShuttle", {
+          reason,
+        });
+      }}
+    />);
+  }
+
+  children.push(
+    <Section title="Emergency Shuttle">
+      {emergencyShuttleButton}
+
+      {shuttleCalledPreviously
+        ? (
+          shuttleLastCalled
+            ? (
+              <Box>
+                Most recent shuttle call/recall traced to:
+                {" "}<b>{shuttleLastCalled}</b>
+              </Box>
+            )
+            : <Box>Unable to trace most recent shuttle/recall signal.</Box>
+        )
+        : null}
```

Comment:
Should be slightly cleaner than ternaries, which are heavily linted atm.

```suggestion
      {!!shuttleCalledPreviously && (
        shuttleLastCalled && (
          <Box>
            Most recent shuttle call/recall traced to:
            {" "}<b>{shuttleLastCalled}</b>
          </Box>
        ) || (
          <Box>Unable to trace most recent shuttle/recall signal.</Box>
        )
      )}
```